test/distributed/test_device_mesh.py

```diff
@@ -1051,6 +1051,26 @@ def test_unflatten_mesh_3d(self):
         )
         w.wait()
 
+    @with_comms
+    def test_unflatten_mesh_3d_with_pg_cache(self):
+        # Turn on gate for not saving PG names for device mesh when it comes to torch.save.
+        # This also turns on pg cache
+        DeviceMesh.decouple_backend_at_save = True
+        # Test unflatten from a dummy world mesh, which is the case we need for Expert Parallelism(EP).
+        global_mesh = init_device_mesh(
+            self.device_type,
+            (8,),
+            mesh_dim_names=("world",),
+        )
+        non_ep_mesh = global_mesh._unflatten(0, (2, 2, 2), ("dp", "cp", "tp"))
+        ep_mesh = global_mesh._unflatten(0, (2, 2, 2), ("dp", "ep", "ep_tp"))
```

Comment:
What's the difference between these two lines vs a user giving multiple names to a dimension? Or a complex name such as "cp_or_ep"?